octue/cli.py

```diff
@@ -123,18 +128,91 @@ def run(app_dir, data_dir, config_dir, input_dir, output_dir, twine):
         show_twined_logs=global_cli_context["show_twined_logs"],
     )
 
+    input_values, input_manifest, children = set_unavailable_strand_paths_to_none(
+        twine,
+        (
+            ("input_values", os.path.join(input_dir, VALUES_FILENAME)),
+            ("input_manifest", os.path.join(input_dir, MANIFEST_FILENAME)),
+            ("children", os.path.join(config_dir, CHILDREN_FILENAME)),
+        ),
+    )
+
     analysis = runner.run(
         app_src=app_dir,
         analysis_id=global_cli_context["analysis_id"],
-        input_values=os.path.join(input_dir, VALUES_FILENAME),
-        input_manifest=os.path.join(input_dir, MANIFEST_FILENAME),
+        input_values=input_values,
+        input_manifest=input_manifest,
+        children=children,
         output_manifest_path=os.path.join(output_dir, MANIFEST_FILENAME),
         skip_checks=global_cli_context["skip_checks"],
     )
     analysis.finalise(output_dir=output_dir)
     return 0
 
 
+@octue_cli.command()
+@click.option(
+    "--app-dir",
+    type=click.Path(),
+    default=".",
+    show_default=True,
+    help="Directory containing your source code (app.py)",
+)
+@click.option(
+    "--data-dir",
+    type=click.Path(),
+    default=".",
+    show_default=True,
+    help="Location of directories containing configuration values and manifest.",
+)
+@click.option(
+    "--config-dir",
+    type=click.Path(),
+    default=None,
+    show_default=True,
+    help="Directory containing configuration (overrides --data-dir).",
+)
+@click.option("--server-name", type=click.STRING, help="The name to give to the server.")
+@click.option("--twine", type=click.Path(), default="twine.json", show_default=True, help="Location of Twine file.")
+def start(app_dir, data_dir, config_dir, service_name, twine):
+    """ Start the service as a server to be asked questions by other services. """
+    config_dir = config_dir or os.path.join(data_dir, FOLDER_DEFAULTS["configuration"])
+    twine = Twine(source=twine)
+
+    runner = Runner(
+        twine=twine,
+        configuration_values=os.path.join(config_dir, VALUES_FILENAME),
+        configuration_manifest=os.path.join(config_dir, MANIFEST_FILENAME),
+        log_level=global_cli_context["log_level"],
+        handler=global_cli_context["log_handler"],
+        show_twined_logs=global_cli_context["show_twined_logs"],
+    )
+
+    children = set_unavailable_strand_paths_to_none(
+        twine, (("children", os.path.join(config_dir, CHILDREN_FILENAME)),)
+    )[0]
+
+    run_function = functools.partial(
+        runner.run, app_src=app_dir, children=children, skip_checks=global_cli_context["skip_checks"],
+    )
+
+    service = Service(name=service_name, run_function=run_function)
+    service.serve()
+
+
+def set_unavailable_strand_paths_to_none(twine, strands):
```

Comment:
This is called further up in the file, so I expected to see it above the class that uses it. Are you aware on any conventions in this respect?

Also, we keep hitting this slight bump of having to tease out None values from the strands. We should probably look to implement a universal method on the Twine class to handle this consistently everywhere. What do you think?

Reply:
There are different conventions on this but the one I've been informally using is that the main thing (usually a public class) for a module is at the top of the file followed by things that are private or of less prominence in order of decreasing prominence. I've also been using a similar convention for ordering methods in a class: `__init__`, properties, other dunder-methods, public methods, then private methods.

I think that would be a good idea although is it `Twine`'s problem or the thing using `Twine`?

Reply:
I've realised that I'm not doing this in tests. Shall we have a chat at some point to set a convention and stick to it?